### DISCUSSION TITLE
How to override typography defaults for specific screen sizes?

### DISCUSSION THREAD

**Discussion by doompadee** [tailwindlabs/tailwindcss#11444] [Help] [answered]
Howdy,

I'm trying to adjust the bottom margin default of headings for bigger screen sizes. According to the documentation I would think the following should do:

``` javascript

const round = (num) =>
  num
    .toFixed(7)
    .replace(/(\.[0-9]+?)0+$/, '$1')
    .replace(/\.0$/, '');
const em = (px, base) => `${round(px / base)}em`;

module.exports = {
  theme: {
    extend: {
      typography: {
        DEFAULT: {
          css: {
            // custom rules for DEFAULT work
          },
        },
        xl: {
          css: {
            // custom rules 
            h1: {
              marginBottom: em(40, 20),
              backgroundColor: 'yellow',
            },
          }
        },
      },
    },
  },
  plugins: [
    require('@tailwindcss/typography'),
  ],
},
```

But to no avail. Overriding DEFAULT values works, but for specific screen sizes, nothing happens. Am I doing something wrong?

I'm using Tailwind 3.3.2 and Typography 0.5.0. Thanks.

**Comment by wongjn** [comment, marked as answer]
Ensure you're applying the `prose-xl` class at the screen size you'd like to it to apply, e.g for `lg:`

```html
<div class="prose lg:prose-xl">
```

**Reply by doompadee** [reply]
Thanks for the quick response. That works, but is unfortunately not what I want.

I was trying to use the defaults, but overriding just one setting for bigger screens and thought this could be achieved directly in the configuration file (which would be more concise than using media queries in the css file).

**Reply by wongjn** [reply]
You could use the element modifier variants along with the media query variant like:

```html
<div class="prose lg:prose-h1:mb-10 lg:prose-h1:bg-yellow-500">
```

**Reply by doompadee** [reply]
Cool, yes, that indeed yields the desired result. Thanks a bunch!

**Comment by dcapitator** [comment]
```
  typography: (theme: any) => ({
        DEFAULT: {
          css: {
            h1: {
              fontSize: {
                sm: theme('fontSize.4xl'),
                md: theme('fontSize.5xl'),
                lg: theme('fontSize.6xl'),
                xl: theme('fontSize.7xl'),
              },
              fontFamily: theme('fontFamily.sadana'),
            },
            h2: {
              fontFamily: theme('fontFamily.myriad'),
            },
            h3: {
              fontFamily: theme('fontFamily.myriad'),
            },
            'blockquote p:first-of-type::before': { content: 'none' },
            'blockquote p:first-of-type::after': { content: 'none' },
          },
        },
      }),
```

is it possible to have different fontsize for diff screen sizes overwritten all in default typography ??
